### PR TITLE
Iter3 order transaction xb

### PR DESCRIPTION
**Pull request by hxb2016**
# Summary

order transaction xb
# Development Deployment

### Steps to Deploy
- Navigate to the Dev App on Heroku: https://dashboard.heroku.com/apps/trackr-dev/deploy/github
- Go down to **Manual deploy**
- Select your branch from the **Choose a branch to deploy** dropdown
- Click **Deploy Branch**
- Et voila! Your code is route to run in development (assuming your build succeeds).

# Testing

Test: https://trackr-dev.herokuapp.com
image

## Automated

# Code Reviews Checklist
- [X] Manageability
- [X] Architecture
- [X] Maintainability
- [X] Correctness
- [X] Invalid input/states
- [X] Usability
- [X] Reusability
- [X] Object-Oriented Analysis and Design (OOAD) Principles

Reference for the list can be found [here](https://www.codegrip.tech/productivity/the-ultimate-code-review-checklist/?utm_source=website&utm_medium=blog&utm_campaign=best-practices-for-code-review-process)

### CODE REVIEW

File: src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java [hunk 1]
```diff
@@ -22,106 +26,139 @@
...
+
+        List<Transaction> transactions = transactionRepository.findByBankAccountIdInAndStatus(bankAccountIds, Transaction.TRANSACTION_STATUS.VALID);
+        transactions.sort((t1, t2) -> {
+            long dateDifference = getMillisecond(t1.getTime()) - getMillisecond(t2.getTime());
+            if (dateDifference == 0) {
+                return 0;
+            } else {
+                return dateDifference > 0 ? -1 : 1;
+            }
+        });
```
Comment (jdorancy-bu) [inline src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java:144]: Order By Time DESC in the JPA query is a better approach.
```java
findByBankAccountIdInAndStatusOrderByTimeDesc()
```

File: src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java [hunk 2]
```diff
@@ -22,106 +26,139 @@
...
+     */
+    private long getMillisecond(String date) {
+        Calendar calendar = Calendar.getInstance();
+        try {
+            calendar.setTime(new SimpleDateFormat("MM/dd/yyyy").parse(date));
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return calendar.getTimeInMillis();
+    }
```
Comment (jdorancy-bu) [inline src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java:163]: We don't need let's remove.

Comment (hxb2016) [reply src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java]: done

File: src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java (hunk 1 above)
Comment (hxb2016) [reply src/main/java/edu/bu/metcs673/trackr/transaction/TransactionServiceImpl.java]: OK, have done
thanks so much!

**Review by jdorancy-bu — APPROVED** [review]
🚀

**Pull request merged**